src/components/app/ProfileViewer.js

```diff
@@ -53,6 +53,34 @@ type DispatchProps = {|
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class ProfileViewer extends PureComponent<Props> {
+  manageBeforeUnloadListener() {
+    const { isUploading } = this.props;
+    if (isUploading) {
+      window.addEventListener(
+        ('beforeunload': BeforeUnloadEvent),
+        this.handleUnload
+      );
+    } else {
+      window.removeEventListener('beforeunload', () => this.handleUnload);
+    }
+  }
+
+  handleUnload = event => {
```

Comment:
I think this is where you should type `event: BeforeUnloadEvent` :-)

In line 60 you tried to coerce a string to an Event object, and of course Flow doesn't accept that.